framework/src/timeintegrators/ActuallyExplicitEuler.C

```diff
@@ -0,0 +1,236 @@
+
+  return params;
+}
+
+/**
+ * Helper class to apply preconditioner
+ */
+class LumpedPreconditioner : public Preconditioner<Real>
+{
+public:
+  LumpedPreconditioner(const NumericVector<Real> & diag_inverse)
+    : Preconditioner(diag_inverse.comm()), _diag_inverse(diag_inverse)
+  {
+  }
+
+  virtual void init() override
+  {
+    // No more initialization needed here
+    _is_initialized = true;
+  }
+
+  virtual void apply(const NumericVector<Real> & x, NumericVector<Real> & y) override
+  {
+    y.pointwise_mult(_diag_inverse, x);
+  }
+
+protected:
+  /// The inverse of the diagonal of the lumped matrix
+  const NumericVector<Real> & _diag_inverse;
+};
+
+ActuallyExplicitEuler::ActuallyExplicitEuler(const InputParameters & parameters)
+  : TimeIntegrator(parameters),
+    MeshChangedInterface(parameters),
+    _solve_type(getParam<MooseEnum>("solve_type")),
+    _explicit_residual(_nl.addVector("explicit_residual", false, PARALLEL)),
+    _explicit_euler_update(_nl.addVector("explicit_euler_update", true, PARALLEL)),
+    _mass_matrix_diag(_nl.addVector("mass_matrix_diag", false, PARALLEL))
+{
+  _Ke_time_tag = _fe_problem.getMatrixTagID("TIME");
+
+  // Try to keep MOOSE from doing any nonlinear stuff
+  _fe_problem.solverParams()._type = Moose::ST_LINEAR;
+
+  if (_solve_type == LUMPED || _solve_type == LUMP_PRECONDITIONED)
+    _ones = &_nl.addVector("ones", false, PARALLEL);
+}
+
+void
+ActuallyExplicitEuler::initialSetup()
+{
+  meshChanged();
+}
+
+void
+ActuallyExplicitEuler::init()
+{
+}
+
+void
+ActuallyExplicitEuler::preSolve()
+{
+}
+
+void
+ActuallyExplicitEuler::computeTimeDerivatives()
+{
+  _u_dot = *_solution;
+  _u_dot -= _solution_old;
+  _u_dot *= 1 / _dt;
+  _u_dot.close();
+
+  _du_dot_du = 1.0 / _dt;
+}
+
+void
+ActuallyExplicitEuler::solve()
+{
+  // This ensures that all the Output objects in the OutputWarehouse
+  // have had solveSetup() called, and sets the default solver
+  // parameters for PETSc.
+  _fe_problem.initPetscOutput();
+
+  auto & es = _fe_problem.es();
+
+  auto & nonlinear_system = _fe_problem.getNonlinearSystemBase();
+
+  auto & libmesh_system = dynamic_cast<NonlinearImplicitSystem &>(nonlinear_system.system());
+
+  auto & mass_matrix = *libmesh_system.matrix;
+
+  _current_time = _fe_problem.time();
+
+  // Set time back so that we're evaluating the interior residual at the old time
+  _fe_problem.time() = _fe_problem.timeOld();
+
+  libmesh_system.update();
+
+  // Must compute the residual first
+  _explicit_residual.zero();
+  _fe_problem.computeResidual(*libmesh_system.current_local_solution, _explicit_residual);
+
+  // The residual is on the RHS
+  _explicit_residual *= -1.;
+
+  // Compute the mass matrix
+  _fe_problem.computeJacobianTag(*libmesh_system.current_local_solution, mass_matrix, _Ke_time_tag);
+
+  // Still testing whether leaving the old update is a good idea or not
+  // _explicit_euler_update = 0;
```

Comment:
Are these 2 lines a left-over from the development?

Reply:
They are left there intentionally - because I still may change it (like in a future PR).  Still evaluating... and I want to see what happens with real problems.